test/org/sagebionetworks/bridge/models/reports/ReportDataKeyTest.java

```diff
@@ -25,17 +26,32 @@ public void constructParticipantKey() {
                 .withReportType(ReportType.PARTICIPANT).withIdentifier("report").build();
 
         assertEquals("healthCode:report:api", key.getKeyString());
+        assertEquals("api:PARTICIPANT", key.getIndexKeyString());
         assertEquals("healthCode", key.getHealthCode());
         assertEquals("report", key.getIdentifier());
         assertEquals(ReportType.PARTICIPANT, key.getReportType());
     }
     
     @Test
+    public void canIncludeDateValidation() {
+        try {
+            // This is tested in tests for the validator, but date is actually validated in the 
+            // build method.
+            new ReportDataKey.Builder().validateWithDate(null).build();
```

Comment:
We need a success case test for validateWithDate().

Also, since you moved some of the validation logic to the builder, we need a test to validate that the builder validates the ReportDataKey.